pandas/core/frame.py

```diff
@@ -6892,56 +6892,93 @@ def cov(self, min_periods=None):
+            * 'all' to compute correlation between all possible pairs of rows
+              or columns
         drop : boolean, default False
             Drop missing indices from result, default returns union of all
+        method : {'pearson', 'kendall', 'spearman'}
+            * pearson : standard correlation coefficient
+            * kendall : Kendall Tau correlation coefficient
+            * spearman : Spearman rank correlation
+        min_periods : int, optional
+            Minimum number of observations required per pair of columns
+            to have a valid result. Currently only available for pearson
+            and spearman correlation
+
+        Returns
+        -------
+        y : DataFrame
 
         Returns
         -------
-        correls : Series
+        correl : Series (how='pairwise') or DataFrame (how='all')
         """
+
         axis = self._get_axis_number(axis)
         this = self._get_numeric_data()
 
         if isinstance(other, Series):
-            return this.apply(other.corr, axis=axis)
+            return this.apply(other.corr, axis=axis,
+                              method=method, min_periods=min_periods)
 
-        other = other._get_numeric_data()
+        if not isinstance(other, DataFrame):
+            raise TypeError("'other' parameter should be a DataFrame "
+                            "or a Series object")
 
-        left, right = this.align(other, join='inner', copy=False)
+        other = other._get_numeric_data()
 
-        # mask missing values
-        left = left + right * 0
-        right = right + left * 0
+        if isinstance(this.columns, MultiIndex) \
+                or isinstance(other.columns, MultiIndex):
+            raise TypeError("MultiIndex is not supported")
 
         if axis == 1:
-            left = left.T
-            right = right.T
-
-        # demeaned data
-        ldem = left - left.mean()
-        rdem = right - right.mean()
+            this = this.transpose()
+            other = other.transpose()
+
+        if how == 'all':
+            corr = np.zeros((this.shape[1], other.shape[1]))
+
+            for i, col1 in enumerate(this.columns):
+                for j, col2 in enumerate(other.columns):
+                    corr[i, j] = this.loc[:, col1].corr(
+                        other.loc[:, col2],
+                        method=method, min_periods=min_periods)
+            return DataFrame(data=corr, index=this.columns,
+                             columns=other.columns)
+
+        elif how == 'pairwise':
+            index = []
+            corr = []
+            for col in this.columns:
+                if col in other.columns:
+                    index.append(col)
+                    corr.append(this.loc[:, col].corr(other.loc[:, col],
+                                method=method, min_periods=min_periods))
+
+            correl = Series(data=corr, index=index)
+
+            if not drop:  # add missing columns to the resulting Series
```

Comment:
huh? what are you trying to do here? you are fighting pandas by doing this

Reply:
I'm adding missing columns in case `drop` param set to `False`. Not sure I understand you.